roles/deprovision-keycloak-apb/tasks/cleanup-keycloak.yml

```diff
@@ -47,29 +47,65 @@
     var: kc_auth_response
     verbosity: 2
 
-- name: Delete client {{ kc_configmap_user.stdout }} in realm {{ kc_configmap_realm.stdout }}
+- name: Fail due to invalid credentials
+  fail: msg="Failed to get access token due to invalid credentials"
+  when: kc_auth_response.status == 401
+
+- name: Fail due to lack of authorization
+  fail: msg="Failed to get access token. The user {{ kcdev_username.stdout }} is unauthorized to perform this action"
+  when: kc_auth_response.status == 403
+
+- name: Get {{ kc_configmap_user.stdout }} and {{ kc_configmap_user.stdout }}-bearer clients in realm {{ kc_configmap_realm.stdout }}
```

Comment:
Wondering if this does anything if we changed the delete to be ok on a 404?

Reply:
Name should be Delete client if available in realm

Reply:
Yep, agree with Craig. You dont need to do Get and then Delete. You can do just Delete and `status_code: 200, 404` that means if that is 404 we still ok that client is not already there